packages/arb-avm-cpp/data_storage/src/arbcore.cpp

```diff
@@ -1801,11 +1797,17 @@ void ArbCore::operator()() {
                 ValueCache logs_cache{1, 0};
                 if (logs_cursors[i].status == DataCursor::REQUESTED) {
                     ReadTransaction tx(data_storage);
-                    handleLogsCursorRequested(tx, i, logs_cache);
+                    auto status = handleLogsCursorRequested(tx, i, logs_cache);
+                    if (!status) {
+                        // error already logged
+                        arbcore_abort = true;
+                        break;
+                    }
                 }
             }
 
-            if (machine_idle && message_data_status != MESSAGES_READY) {
+            if (!arbcore_abort && machine_idle &&
```

Comment:
Could we just check for `arbcore_error` here? I don't see anywhere we set `arbcore_abort` that isn't an error. Also if we switch to just returning on error as suggested above, we might not need to check either

Reply:
will handle in separate PR with above.